### PR TITLE
Add OneHot layer for generating one hot embedding for zero-based integer features

### PR DESCRIPTION
**Pull request by nhanitvn**

### CODE REVIEW

File: keras/layers/embeddings.py [hunk 1]
```diff
@@ -149,3 +149,49 @@ def get_config(self):
                   'dropout': self.dropout}
         base_config = super(Embedding, self).get_config()
         return dict(list(base_config.items()) + list(config.items()))
+
+
+class OneHot(Layer):
+    '''Turn positive integers (indexes) into dense dummy vectors of fixed size
```
Comment (fchollet) [inline keras/layers/embeddings.py]: Dense dummy vectors?

**Comment by nhanitvn** [comment]
@fchollet Thanks for reviewing the PR. I wrote about the reason for a dedicated layer in https://github.com/fchollet/keras/issues/3680. Basically, we may want a flexibility in encoding categorical features during network construction.

File: keras/layers/embeddings.py (hunk 1 above)
Comment (nhanitvn) [inline keras/layers/embeddings.py]: Ah, may be "one-hot" is a more appropriate term.

**Comment by fchollet** [comment]
This the layer is stateless (no weights) it is preferable to use `Lambda(K.one_hot)` IMO.

**Comment by nhanitvn** [comment]
Agree @fchollet. Thanks for your comments. At least it is a good experience writing a custom Keras layer, :).

**Pull request closed without merging**